Missing "deployments" folder

When executing: $build/bin/templating-device --ipam-range 10.10.0.0/16     --number-of-replicas 500     --input-dir templates     --output-dir deployment     --retry-period 15

panic: open deployment/net-attach-def.yaml: no such file or directory

Dir "deployment" needs to be either precreated or created on demand.

--- a/cmd/templating-device.go
+++ b/cmd/templating-device.go
@@ -94,6 +94,9 @@ func (td templatingDevice) preRenderedTemplate(inputFile string) *template.Templ
 }
 
 func (td templatingDevice) spit(outputDir string) error {
+	if err := os.MkdirAll(outputDir, 0700); err != nil {
+		return fmt.Errorf("failed to ensure dir %q exists: %w", outputDir, err)
+	}
 	for templatePath, templateData := range td.templatingEngine {
 		preRenderedTemplate, ok := td.results[templatePath]
 		if !ok {

